### ISSUE TITLE
`Layout/LineLength` safe autocorrect results in invalid code

### ISSUE THREAD

**Issue by a-lavis** [rubocop/rubocop#13096]
Sometimes the `Layout/LineLength` cop, which is marked as a safe autocorrect, corrects to invalid ruby code which cannot be parsed.

## Expected behavior

When running `rubocop -a foo.rb` where `foo.rb` is the following file:

```ruby
# frozen_string_literal: true

puts ['foo'] + ['bar'].map { |x| x.upcase }, 'ok'
```

And I have the following `.rubocop.yml`:

```yml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.3

Layout/LineLength:
  Max: 48
```

I expect it to not result in invalid code.

## Actual behavior

It produces the following file:

```ruby
# frozen_string_literal: true

puts ['foo'] + ['bar'].map do |x|
  x.upcase
end, 'ok'
```

Running `ruby foo.rb` produces the following output:
```
foo.rb: --> foo.rb
unexpected write target
> 3  puts ['foo'] + ['bar'].map do |x|
> 5  end, 'ok'

foo.rb:5: syntax error, unexpected ',', expecting end-of-input (SyntaxError)
end, 'ok'
```

## Steps to reproduce

Follow the above steps in the "Expected behavior section"

## RuboCop version

```
> rubocop -V
1.65.1 (using Parser 3.3.4.1, rubocop-ast 1.32.0, running on ruby 3.3.4) [arm64-darwin23]
```

**Comment by a-lavis** [comment]
I've found two other programs that can reproduce similar bugs.  I'm not sure if they should be their own GitHub issues, so I'm just posting them here for now.

---

### Extra example 1:

With this config:
```yml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.3

Layout/LineLength:
  Max: 44
```

This file

```ruby
# frozen_string_literal: true

def foo
  puts 'foo'
  true
end

['bar'].map { |x| x.upcase } unless foo # foo
puts { |x| x.upcase }
```

safe autocorrects to:

```ruby
# frozen_string_literal: true

def foo
  puts 'foo'
  true
end

# foo
['bar'].map do |x|
  x.upcase
end
 unless foo puts { |x| x.upcase }
```

This is syntactically invalid ruby code, like the example in the PR description.  However it seems to be a slightly different bug than the one in the PR description, so I thought I'd include it just in case.

---

### Extra example 2:

With this config:
```yml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.3

Layout/LineLength:
  Max: 53
```

This file

```ruby
# frozen_string_literal: true

def foo
  puts 'foo'
  true
end

"baz #{['bar'].map { |x| x.upcase }}" unless foo # foo
puts { |x| x.upcase }
```

safe autocorrects to

```ruby
# frozen_string_literal: true

def foo
  puts 'foo'
  true
end

unless foo(puts { |x| x.upcase })
  "baz #{ # foo
  ['bar'].map do |x|
    x.upcase
  end
}"
end
```

This is syntactically valid Ruby code, but it is meaningfully different from the original.  You can run the original just fine, but when you try to run the autocorrected one you get an error, because RuboCop autocorrected it to pass arguments into the `foo` method, which takes none.

**Comment by dvandersluis** [comment]
In the first case, `Layout/LineLength` on its own doesn't actually make invalid code:

```ruby
puts map { |x|
 x.upcase }, 'ok'
```

However, then `Style/BlockDelimiters` changes the braces to `do...end`, which makes it a syntax error:

```ruby
puts ['foo'] + ['bar'].map do |x|
 x.upcase end, 'ok'
```

A solution would when `Layout/LineLength` splits inside arguments of a `send` node, it wraps them in parentheses if they aren't already, because this is valid:

```
puts(['foo'] + ['bar'].map do |x|
 x.upcase end, 'ok')
```

In fact, if you start with parenthesized arguments:

```ruby
puts(['foo'] + ['bar'].map { |x| x.upcase }, 'ok')
```

RuboCop full autocorrection results in valid, nice-looking code:

```ruby
puts(
  ['foo'] + ['bar'].map(&:upcase), 'ok'
)
```

**Comment by dvandersluis** [comment]
Extra example 1 is a bit different.

If we start with (slightly simplified from your example):

```ruby
['bar'].map { |x| x.upcase } unless foo # foo
```

Only running `Layout/LineLength` on it results in:

```ruby
['bar'].map { |x|
 x.upcase } unless foo # foo
```

Next, running `Style/BlockDelimiters`:

```ruby
# foo
['bar'].map do |x|
 x.upcase end
 unless foo 
```

In this case, it's `Style/BlockDelimiters` that has produced an invalid autocorrect, so this is a separate issue.

**Comment by dvandersluis** [comment]
Note: extra example 2 is also covered by the fix in #13349.

**Comment by dvandersluis** [comment]
@a-lavis the original issue is now addressed in #13499.
